Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/CommonComponents/LocationDrawingsButton.tsx

```diff
@@ -0,0 +1,160 @@
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  01/06/2024 - Collins Self
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+import React from 'react';
+import { BtnDropdown, GenericController, LoadingScreen, ServerErrorIcon, ToolTip } from '@gpa-gemstone/react-interactive';
+import LocationDrawingsModal from './LocationDrawingsModal';
+import { OpenXDA } from '@gpa-gemstone/application-typings';
+import { CrossMark } from '@gpa-gemstone/gpa-symbols';
+
+interface LocationDrawingsButtonProps {
+    Locations: OpenXDA.Types.Location[];
+}
+
+type DropDownOption = {
+    Label: JSX.Element | string,
+    Callback: () => void,
+    Disabled: boolean
+    ToolTipContent: JSX.Element,
+    ShowToolTip: boolean,
+    ToolTipLocation: ('top' | 'bottom' | 'left' | 'right'),
+    Key: string | number
+}
+
+const LocationDrawingsButton: React.FC<LocationDrawingsButtonProps> = (props) => {
+    const [hover, setHover] = React.useState<'none' | 'drawings'>('none');
+    const [pageState, setPageState] = React.useState<"loading" | "error" | "idle">("idle");
+    const [selectedLocation, setSelectedLocation] = React.useState<OpenXDA.Types.Location>();
+    const multipleLocations = React.useMemo(() => props.Locations.length > 1, [props.Locations]);
+    const [showDrawingsModal, setShowDrawingsModal] = React.useState<boolean>(false);
+    const [locationOptions, setLocationOptions] = React.useState<DropDownOption[]>([]);
+    const LocationDrawingController = new GenericController(`${homePath}api/LocationDrawing`, "Name", true);
+
+    const isValid = (location: OpenXDA.Types.Location, drawingData) => {
+        let e = "";
+
+        if (!location
+            || (location.Alias == ""
+                && location.Description == ""
+                && location.ID == 0
+                && location.Latitude == null
+                && location.LocationKey == ""
+                && location.Longitude == null
+                && location.Name == ""))
+            e = 'No locations have been set.';
+        else if (drawingData.TotalRecords == 0)
+            e = 'No drawings associated with location.';
+        return e;
+    }
+
+    React.useEffect(() => { // Generates the map of errors for each location
+        setPageState('loading');
+
+        const handles = props.Locations.map((location, i) => {
+            setLocationOptions([]);
```

Comment:
Each location is going to fight each other on what actually happens to this setLocationOptions, only one location will win.
You should be setting this when you set the page state later.